Add NNC support for aten::hardtanh (a hot operation in mobilenet v2/v3)

huiguoo opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#52394 Add NNC support for aten::hardtanh (a hot operation in mobilenet v2/v3)**

Differential Revision: [D26497856](https://our.internmc.facebook.com/intern/diff/D26497856)

huiguoo committed:
Add NNC support for aten::hardtanh (a hot operation in mobilenet v2/v3)

[ghstack-poisoned]

bertmaher reviewed (approved):
Awesome, wow!  Please clang-format (there's a script to do it automatically, tools/clang_format_all.py).

huiguoo committed:
Update on "Add NNC support for aten::hardtanh (a hot operation in mobilenet v2/v3)"


Differential Revision: [D26497856](https://our.internmc.facebook.com/intern/diff/D26497856)

[ghstack-poisoned]

ZolotukhinM commented:
It would be cool to see if that gets fused in torch benchmarks and improves perf. For instance, we know that mobilenet has hardtanh (but inplace version) - could you check if it starts getting fused? To remove inplace you could try running this pass
https://github.com/pytorch/pytorch/blob/c7a70eec1b1a8aaac1f5659f688cd625c4a67831/torch/csrc/jit/passes/remove_inplace_ops.h#L10
in this function:
https://github.com/pytorch/pytorch/blob/c7a70eec1b1a8aaac1f5659f688cd625c4a67831/torch/csrc/jit/runtime/profiling_graph_executor_impl.cpp#L192

huiguoo commented:
Great, thank you @ZolotukhinM ! Will try it out!
